merkle/hstar2.go

```diff
@@ -83,19 +87,51 @@ func (s *HStar2) HStar2Nodes(prefix []byte, subtreeDepth int, values []*HStar2Le
 			glog.Infof("  %x: %x", v.Index.Bytes(), v.LeafHash)
 		}
 	}
+	combine := func(depth int, offset *big.Int, lhs, rhs []byte) ([]byte, error) {
+		h := s.hasher.HashChildren(lhs, rhs)
+		if err := s.set(offset, depth, h, set); err != nil {
+			return nil, err
+		}
+		return h, nil
+	}
+	return s.run(prefix, subtreeDepth, values, get, combine)
+}
+
+// Prefetch does a dry run of HStar2 algorithm, and reports all Merkle tree
+// nodes that it needs through the passed-in visit function. Note that the
+// return value of the visit function is ignored, unless it is an error.
```

Comment:
Consider whether a different function signature for the visit argument is better than SparseGetNodeFunc. Having a return argument that isn't used is a little messy, and I can't see too many use-cases where clients would want to pass the same SparseGetNodeFunc into this as the other places it is used: if a prefetch is useful then the visit will likely always be a Node ID accumulator.

Reply:
Done.